opm/output/eclipse/Summary.cpp

```diff
@@ -29,223 +29,103 @@
-    WOPR,
-    WOPRH,
-    WOPT,
-    WOPTH,
-    WTHP,
-    WTHPH,
-    WWCT,
-    WWCTH,
-    WWIR,
-    WWIRH,
-    WWIT,
-    WWITH,
-    WWPR,
-    WWPRH,
-    WWPT,
-    WWPTH,
-    GWPT,
-    GOPT,
-    GGPT,
-    GWPR,
-    GOPR,
-    GLPR,
-    GGPR,
-    GWIR,
-    GWIT,
-    GGIR,
-    GGIT,
-    GGIRH,
-    GWIRH,
-    GOIRH,
-    GGITH,
-    GWITH,
-    GWPRH,
-    GOPRH,
-    GGPRH,
-    GLPRH,
-    GWPTH,
-    GOPTH,
-    GGPTH,
-    GLPTH,
-    GWCT,
-    GGOR,
-    UNSUPPORTED,
-};
+/* Some numerical value with its unit tag embedded to enable caller to apply
+ * unit conversion. This removes a ton of boilerplate.
+ *
+ * Some arithmetic operations are supported, which also codifies the following
+ * assumptions:
+ * * Division by zero, such as in gas/oil ratios without oil, shall default to 0
+ * * Division typically happen to calculate ratios, in which case unit
+ *   conversion is a no-op
+ * * When performing an operation on two operands, the left operand's unit is
+ *   applied, meaning arithmetic is **not** associative with respect to units.
+ */
+struct uv {
+    double value;
+    UnitSystem::measure unit;
 
-const std::map< std::string, E > keyhash = {
-    { "WBHP",  E::WBHP },
-    { "WBHPH", E::WBHPH },
-    { "WGIR",  E::WGIR },
-    { "WGIRH", E::WGIRH },
-    { "WGIT",  E::WGIT },
-    { "WGITH", E::WGITH },
-    { "WGOR",  E::WGOR },
-    { "WGORH", E::WGORH },
-    { "WGLR",  E::WGLR },
-    { "WGLRH", E::WGLRH },
-    { "WGPR",  E::WGPR },
-    { "WGPRH", E::WGPRH },
-    { "WGPT",  E::WGPT },
-    { "WGPTH", E::WGPTH },
-    { "WLPR",  E::WLPR },
-    { "WLPRH", E::WLPRH },
-    { "WLPT",  E::WLPT },
-    { "WLPTH", E::WLPTH },
-    { "WOIR",  E::WOIR },
-    { "WOIRH", E::WOIRH },
-    { "WOIT",  E::WOIT },
-    { "WOITH", E::WOITH },
-    { "WOPR",  E::WOPR },
-    { "WOPRH", E::WOPRH },
-    { "WOPT",  E::WOPT },
-    { "WOPTH", E::WOPTH },
-    { "WTHP",  E::WTHP },
-    { "WTHPH", E::WTHPH },
-    { "WWCT",  E::WWCT },
-    { "WWCTH", E::WWCTH },
-    { "WWIR",  E::WWIR },
-    { "WWIRH", E::WWIRH },
-    { "WWIT",  E::WWIT },
-    { "WWITH", E::WWITH },
-    { "WWPR",  E::WWPR },
-    { "WWPRH", E::WWPRH },
-    { "WWPT",  E::WWPT },
-    { "WWPTH", E::WWPTH },
-    { "GWPT",  E::GWPT },
-    { "GOPT",  E::GOPT },
-    { "GGPT",  E::GGPT },
-    { "GWPR",  E::GWPR },
-    { "GOPR",  E::GOPR },
-    { "GLPR",  E::GLPR },
-    { "GGPR",  E::GGPR },
-    { "GWIR",  E::GWIR },
-    { "GWIT",  E::GWIT },
-    { "GGIR",  E::GGIR },
-    { "GWPRH", E::GWPRH },
-    { "GOPRH", E::GOPRH },
-    { "GGPRH", E::GGPRH },
-    { "GLPRH", E::GLPRH },
-    { "GWPTH", E::GWPTH },
-    { "GOPTH", E::GOPTH },
-    { "GGPTH", E::GGPTH },
-    { "GLPTH", E::GLPTH },
-    { "GGIRH", E::GGIRH },
-    { "GWIRH", E::GWIRH },
-    { "GOIRH", E::GOIRH },
-    { "GGIT",  E::GGIT },
-    { "GWITH", E::GWITH },
-    { "GGITH", E::GGITH },
-    { "GWCT",  E::GWCT },
-    { "GGOR",  E::GGOR },
+    uv operator+( const uv& rhs ) const {
```

Comment:
All of these operators are in my opinion misleading and dangerous. Addition of quantities with different units should not be allowed, that is silently accepted and the lhs unit used for the result. Multiplication should yield a unit equal to the product unit, not the lhs unit. Division should yield the quotient unit, not identity.